### ISSUE TITLE
Stepper showing implementation details of HtDP templates in *SL

### ISSUE THREAD

**Issue by dvanhorn** [racket/htdp#158]
When stepping through programs that have templates, the stepper is revealing some implementation details about templates that are likely confusing to beginning students.

For example, stepping through this BSL program:
```
(define (temp x)
  (... x ...))

(add1 7)
```
transforms the program into:
```
(define lifted.0 1)
(define (temp x)
  (error
   '...
   "expected a finished expression, but found a template"))
(add1 7)
```

**Comment by jbclements** [comment]
okay, the additional regression is repaired by 1beb59a0c5a7cb2b2884c906 . Additionally, I'm making progress on the original bug as well.

**Comment by jbclements** [comment]
I believe this is all now fixed by #30bf5547d67c2e2a95
